rasa/core/training/interactive.py

```diff
@@ -602,16 +630,38 @@ def _slot_history(tracker_dump: Dict[Text, Any]) -> List[Text]:
                 "value": a.get("action")}
                for a in predictions]
 
-    choices = ([{"name": "<create new action>", "value": OTHER_ACTION}] +
-               choices)
+    tracker = await retrieve_tracker(endpoint, sender_id)
+    evts = tracker.get("events", [])
+
+    session_actions_unique = []
+    session_actions_all = [a["name"] for a in _collect_actions(evts)]
+    [session_actions_unique.append(a) for a in session_actions_all if
+     a not in session_actions_unique]
+    old_actions = [action["value"] for action in choices]
+    new_actions = [{"name": action, "value": OTHER_ACTION + action} for action
+                   in session_actions_unique if action not in old_actions]
```

Comment:
there's surely a nicer way to do this? e.g. by using a set on the merged lists of the current actions and the new actions?

Reply:
A set can be used to get an unique actions list out of all current actions, I'll change that. But
we cannot use a set for the new actions that were previously created, since we need to differentiate them from the other actions and give them a newly created value in order to identify them as 'new but not created in same turn' later on.

Reply:
I'll remove the commented lines in the next commit, did't test and clean everything yet